### PR TITLE
enhancement: improve dropdown component quality

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Changelog

```
- Refactor code
- Show label instead of value
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/views/dashboard/settings/views/collectibles/MaxMediaSize.svelte [hunk 1]
```diff
@@ -16,7 +16,7 @@
     import { activeProfile, updateActiveProfileSettings } from '@core/profile/stores'
     import type { IDropdownChoice } from '@core/utils'
 
-    function updateMediaSizeLimit(option): void {
+    function updateMediaSizeLimit(option: IDropdownChoice<number>): void {
```
Comment (maxwellmattryan) [inline packages/desktop/views/dashboard/settings/views/collectibles/MaxMediaSize.svelte:19]: This should be named `onUpdateMediaSizeLimitClick`

File: packages/desktop/views/dashboard/settings/views/general/Language.svelte [hunk 2]
```diff
@@ -4,10 +4,10 @@
     import { SUPPORTED_LOCALES, localize, setLanguage } from '@core/i18n'
     import type { IDropdownChoice } from '@core/utils'
 
-    let languageList: IDropdownChoice[]
+    let languageList: IDropdownChoice<string>[]
     $: languageList = Object.values(SUPPORTED_LOCALES).map((locale) => ({ value: locale, label: locale }))
 
-    function handleLanguage(item): void {
+    function handleLanguage(item: IDropdownChoice<string>): void {
```
Comment (maxwellmattryan) [inline packages/desktop/views/dashboard/settings/views/general/Language.svelte:10]: `onLanguageClick`

File: packages/desktop/views/dashboard/settings/views/profile/Currency.svelte [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
     import { Dropdown, Text } from 'shared/components'
 
-    let currencyList: IDropdownChoice[]
+    let currencyList: IDropdownChoice<MarketCurrency>[]
     $: currencyList = Object.values(MarketCurrency)
         .map((currency) => ({ value: currency, label: currency.toUpperCase() }))
         .sort()
 
-    function handleCurrencySelect(item): void {
+    function handleCurrencySelect(item: IDropdownChoice<MarketCurrency>): void {
```
Comment (maxwellmattryan) [inline packages/desktop/views/dashboard/settings/views/profile/Currency.svelte:13]: `onCurrencySelectClick`

File: packages/desktop/views/dashboard/settings/views/security/AppLock.svelte [hunk 4]
```diff
@@ -4,7 +4,7 @@
     import { activeProfile, DEFAULT_PERSISTED_PROFILE_OBJECT, updateActiveProfileSettings } from '@core/profile'
     import type { IDropdownChoice } from '@core/utils'
 
-    function updateLockTimeout(option): void {
+    function updateLockTimeout(option: IDropdownChoice<number>): void {
```
Comment (maxwellmattryan) [inline packages/desktop/views/dashboard/settings/views/security/AppLock.svelte:7]: `onUpdateLockTimeoutClick`

File: packages/desktop/views/dashboard/settings/views/security/StrongholdPasswordTimeout.svelte [hunk 5]
```diff
@@ -5,7 +5,7 @@
     import { IDropdownChoice, SECONDS_PER_MINUTE } from '@core/utils'
     import { setStrongholdPasswordClearInterval } from '@core/profile-manager'
 
-    function onSelect(option): void {
+    function onSelect(option: IDropdownChoice<number>): void {
```
Comment (maxwellmattryan) [inline packages/desktop/views/dashboard/settings/views/security/StrongholdPasswordTimeout.svelte:8]: `onTimeoutSelectClick`

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have some requested changes - have to review the rest tomorrow :sweat_smile:

**Review by MarkNerdi — APPROVED** [review]
LGTM :fire:

**Pull request merged**